packages/dom/src/dom.js

```diff
@@ -389,13 +389,21 @@ export function placeCaretAtVerticalEdge( container, isReverse, rect, mayUseScro
  * @return {boolean} True if the element is an text field, false if not.
  */
 export function isTextField( element ) {
-	const { nodeName, selectionStart, contentEditable } = element;
-
-	return (
-		( nodeName === 'INPUT' && selectionStart !== null ) ||
-		( nodeName === 'TEXTAREA' ) ||
-		contentEditable === 'true'
-	);
+	try {
+		const { nodeName, selectionStart, contentEditable } = element;
+
+		return (
+			( nodeName === 'INPUT' && selectionStart !== null ) ||
+			( nodeName === 'TEXTAREA' ) ||
+			contentEditable === 'true'
+		);
+	} catch ( error ) {
```

Comment:
Hi @chrisvanpatten thank you for catching and addressing this issue.
It seems the isTextField can be called on safari many times and on safari calling it on non-text inputs and having to catch this exception may be expected/part of the normal flow. Ideally, exceptions should not be used for expected flows.
Do you think checking for a list of input types (text, email etc...) when the node name is equal to input can be a valid solution?

Reply:
@jorgefilipecosta I had thought about that but was concerned the list of valid input types could change from browser-to-browser and be harder to maintain. That said, I'm totally fine with that too.

Might need help with structuring the function… here's what I'd do if it were just for personal code. Does this look okay?

```
if ( element.nodeName === 'TEXTAREA' || element.contentEditable === true ) {
	return true;
}

// List of valid input types for selectionStart via WHATWG spec.
// https://html.spec.whatwg.org/multipage/input.html#concept-input-apply
const validInputTypes = [
	'password',
	'search',
	'tel',
	'text',
	'url',
];

if ( element.nodeName === 'INPUT' && ! includes( validInputTypes, element.type ) ) {
	return false;
}

if ( element.selectionStart !== null ) {
	return true;
}

return false;
```

Reply:
Sorry @chrisvanpatten I missed your ping.  In my option, the logic you propose seems nice and is preferable over using a try catch. Maybe we can make validInputTypes a constant VALID_INPUT_TYPES_SELECTION_START or something similar.

Reply:
I'm fine with either the revised approach or the original proposal. The `try` / `catch` is agreeable only because it seems in this case Safari is not following the standard by throwing, where the standard dictates:

>If this element is an `input` element, and `selectionStart` [does not apply](https://html.spec.whatwg.org/multipage/input.html#do-not-apply) to this element, return `null`.

https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-textarea/input-selectionstart

...in which case we're capturing an unpredictable (non-standard) behavior.

The revised approach recreates the intent of the standard above, though in which case I'd find the condition of `if ( element.selectionStart !== null ) {` to be redundant and unnecessary.